src/sysroots.c

```diff
@@ -37,34 +48,28 @@
 **  name of a library file 'strcat( SyGapRootPaths[i], "lib/init.g" );'  must
 **  be a valid filename.
 */
-enum { MAX_GAP_DIRS = 16 };
-static Char SyGapRootPaths[MAX_GAP_DIRS][GAP_PATH_MAX];
+Obj SyGapRootPaths;
 
 
 /****************************************************************************
 **
-*F  SyFindGapRootFile( <filename>, <buf>, <size> ) . find file in system area
+*F  SyFindGapRootFile( <filename> ) . . . . . . . .  find file in system area
 **
-**  <buf> must point to a buffer of at least <size> characters. This function
-**  then searches for a readable file with the name <filename> in the system
-**  area. If sich a file is found then its absolute path is copied into
-**  <buf>, and <buf> is returned. If no file is found or if <buf> is not big
-**  enough, then <buf> is set to an empty string and NULL is returned.
+**  This function searches for a readable file with the name <filename> in
+**  the system area. If such a file is found then its absolute path is
+**  returned as a string object. If no file is found then NULL is returned.
 */
-Char * SyFindGapRootFile(const Char * filename, Char * buf, size_t size)
+Obj SyFindGapRootFile(const Char * filename)
```

Comment:
Unfortunately changing `SyFindGapRootFile` to return a string object currently clashes with work space restoring as it is called during that when restoring loaded compiled modules (including kernel extensions)